### ISSUE TITLE
TpetraTSQR_SequentialTsqr_[non]contiguousCacheBlocks_MPI_1 tests failing randomly with 'illegal instruction was encountered' in PR `rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release` builds starting 2022-07-20

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#10847]
CC: @trilinos/framework, @trilinos/tpetra

## Internal Issues:

* [TRILINOSHD-155](https://sems-atlassian-son.sandia.gov/jira/servicedesk/customer/portal/7/TRILINOSHD-155)

## Next Action Status

<status-and-or-first-action>

## Description

As shown in [this query](https://trilinos-cdash.sandia.gov/queryTests.php?project=Trilinos&begin=2022-07-01&end=2022-08-14&filtercount=4&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Pull%20Request&field2=buildname&compare2=63&value2=rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables&field3=status&compare3=61&value3=failed&field4=testoutput&compare4=95&value4=an%20illegal%20instruction%20was%20encountered) (click "Shown Matching Output" in upper right) the tests:

* `TpetraTSQR_SequentialTsqr_contiguousCacheBlocks_MPI_1`
* `TpetraTSQR_SequentialTsqr_noncontiguousCacheBlocks_MPI_1`

failed 23 times in the builds:

* `PR-10751-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-305`
* `PR-10767-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-34`
* `PR-10775-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-13`
* `PR-10775-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-387`
* `PR-10777-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-32`
* `PR-10783-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-22`
* `PR-10784-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-19`
* `PR-10784-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-29`
* `PR-10785-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-14`
* `PR-10785-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-23`
* `PR-10791-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-329`
* `PR-10791-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-341`
* `PR-10791-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-366`
* `PR-10796-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-238`
* `PR-10801-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-331`
* `PR-10808-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-343`
* `PR-10808-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-347`
* `PR-10808-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-382`
* `PR-10808-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-389`
* `PR-10808-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-407`
* `PR-10813-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-348`
* `PR-10825-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-403`
* `PR-10825-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-412`

started failing on testing day 2022-07-20 through testing day 2022-08-14.

As shown above, this has impacted so far PRs #10751, #10767, #10775, #10777, #10783, #10784, #10785, #10791, #10801, #10808, #10813, and #10825.

As shown in [this query](https://trilinos-cdash.sandia.gov/queryTests.php?project=Trilinos&begin=2022-07-15&end=2022-08-14&filtercount=4&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Pull%20Request&field2=buildname&compare2=63&value2=rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables&field3=block&field3count=2&field3field1=testname&field3compare1=61&field3value1=TpetraTSQR_SequentialTsqr_contiguousCacheBlocks_MPI_1&field3field2=testname&field3compare2=61&field3value2=TpetraTSQR_SequentialTsqr_noncontiguousCacheBlocks_MPI_1&field4=testname&compare4=61&value4=TpetraTSQR_SequentialTsqr_noncontiguousCacheBlocks_MPI_1), since 2022-07-20 when these tests first started running in this build, that they have run run 74 total times in PR testing.  Therefore, that is good evidence that this test is randomly failing.

**NOTE:** This test failed 23 times out of being run 72 time.  That gives it about a 32% probability of failure!

When this fails, it shows "an illegal instruction was encountered":

```
NodeTsqr verify/benchmark test options:
  - NodeTsqr: SequentialTsqr
  - numRows: 100000
  - numCols: 10
  - numTrials: 10
  - testReal: true
  - testComplex: false
  - cacheSizeHint: 5000
  - contiguousCacheBlocks: false
  - printFieldNames: true
  - printTrilinosTestStuff: true
  - humanReadable: false
  - verbose: false
  - saveMatrices: false
%method,scalarType,numRows,numCols,cacheSizeHint,contiguousCacheBlocks,frobA,absFrobResid,absFrobOrthog
CUSOLVER,float,100000,10,0,false,1.15429,4.47428e-07,6.45184e-07
LAPACK,float,100000,10,0,false,1.15429,2.71137e-07,7.84201e-07
terminate called after throwing an instance of 'std::runtime_error'
  what():  cudaMemcpyAsync(dst, src, n, cudaMemcpyDefault, instance.cuda_stream()) error( cudaErrorIllegalInstruction): an illegal instruction was encountered /scratch/trilinos/workspace/Trilinos_PR_cuda-11.4.2-uvm-off@2/Trilinos/packages/kokkos/core/src/Cuda/Kokkos_CudaSpace.cpp:99
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpiexec noticed that process rank 0 with PID 0 on node ascicgpu050 exited on signal 6 (Aborted).
--------------------------------------------------------------------------
```

Since this test is randomly failing and since this is a new PR build that started running on 2022-07-15, it is impossible to determine what commit may have triggered these failures.

## Current Status on CDash

Run the [above query](https://trilinos-cdash.sandia.gov/queryTests.php?project=Trilinos&begin=2022-07-01&end=2022-08-14&filtercount=4&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Pull%20Request&field2=buildname&compare2=63&value2=rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables&field3=status&compare3=61&value3=failed&field4=testoutput&compare4=95&value4=an%20illegal%20instruction%20was%20encountered) adjusting the "Begin" and "End" dates to match today any other date range or just click "CURRENT" in the top bar to see results for the current testing day.

## Steps to Reproduce

One should be able to reproduce this failure as described in:

* https://github.com/trilinos/Trilinos/wiki/Reproducing-PR-Testing-Errors

For questions about that, please contact @trilinos/framework.

**Comment by bartlettroscoe** [comment]
FYI: This test also single-handily took out the last PR testing iteration for PR #10813 shown https://github.com/trilinos/Trilinos/pull/10813#issuecomment-1204791369 and [here](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&filtercount=3&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-10813-test&field2=buildstarttime&compare2=83&value2=2022-08-08T23%3A43%3A22%20UTC&field3=buildstarttime&compare3=84&value3=2022-08-08T23%3A47%3A13%20UTC).

**Comment by csiefer2** [comment]
We know about this and it is scheduled to get looked at.

**Comment by bartlettroscoe** [comment]
> We know about this and it is scheduled to get looked at.

I did not see an existing GitHub issue for this failure, so I created this.  If someone knows about an existing issue, then it really helps to create an issue so that others can find it.

**Comment by bartlettroscoe** [comment]
Can we disable this test for now in PR builds so that it does not block the merging of PRs?

**Comment by csiefer2** [comment]
No.  Because we can't merge a PR to disable it.

**Comment by bartlettroscoe** [comment]
> No. Because we can't merge a PR to disable it.

No, the disable occurs in the GenConfig *.ini files which don't go through PR testing (but they should).

And even if they did, the PRs would still merge because they would trigger `-DTrilinos_ENABLE_ALL_PACKAGES=ON` (which masks the TriBITS defect causing #10842).

**Comment by csiefer2** [comment]
Well, seeing as I can't merge into the GenConfig repo, I can't exactly do that either.

**Comment by bartlettroscoe** [comment]
> Well, seeing as I can't merge into the GenConfig repo, I can't exactly do that either. Maybe you can.

I will put in an MR to the ini repo to do this.

**Comment by bartlettroscoe** [comment]
I posted:

* https://sems-atlassian-son.sandia.gov/jira/servicedesk/customer/portal/7/TRILINOSHD-164

to have this test disabled in this one PR CUDA build.

**Comment by bartlettroscoe** [comment]
This is resolved by PR #10871.  Putting in review.

**Comment by bartlettroscoe** [comment]
Shoot, I did not look over other failures on this CUDA-11 build showing "an illegal instruction was encountered".  Running the fuller query:

* [All test failures in "Pull Request" build `rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables` showing "an illegal instruction was encountered" from 2022-07-01 through 2022-08-14](https://trilinos-cdash.sandia.gov/queryTests.php?project=Trilinos&begin=2022-07-01&end=2022-08-14&filtercount=4&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Pull%20Request&field2=buildname&compare2=63&value2=rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables&field3=status&compare3=61&value3=failed&field4=testoutput&compare4=95&value4=an%20illegal%20instruction%20was%20encountered).

That shows 26 failures involving the tests:

* `TpetraTSQR_SequentialTsqr_contiguousCacheBlocks_MPI_1`
* `TpetraTSQR_SequentialTsqr_noncontiguousCacheBlocks_MPI_1`

I will update the description [above](https://github.com/trilinos/Trilinos/issues/10847#issue-1333262203) to cover both of these tests.

**Comment by bartlettroscoe** [comment]
Actually, the correct statistic is how may builds get taken out by one of these two tests.   [This query](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&begin=2022-07-15&end=2022-08-14&filtercount=4&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Pull%20Request&field2=buildname&compare2=63&value2=rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables&field3=subprojects&compare3=93&value3=Tpetra&field4=testspassed&compare4=43&value4=0) shows that 80 of these builds have been run that included Tpetra tests while [this query](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&begin=2022-07-15&end=2022-08-14&filtercount=4&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Pull%20Request&field2=buildname&compare2=63&value2=rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables&field3=subprojects&compare3=93&value3=Tpetra&field4=testsfailed&compare4=43&value4=0) shows that 24 of those had at least one Tpetra test failure.  So that gives about a 30% failure rate.

And in some iterations, both of these tests fail like in:

* [PR-10751-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-305](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&parentid=335985&filtercount=3&showfilters=1&field1=groupname&compare1=61&value1=Pull%20Request&field2=subprojects&compare2=93&value2=Tpetra&field3=testsfailed&compare3=43&value3=0&filtercombine=and)
* [PR-10784-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-29](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&parentid=305011&filtercount=3&showfilters=1&field1=groupname&compare1=61&value1=Pull%20Request&field2=subprojects&compare2=93&value2=Tpetra&field3=testsfailed&compare3=43&value3=0&filtercombine=and)
* [PR-10775-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-13](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&parentid=298736&filtercount=3&showfilters=1&field1=groupname&compare1=61&value1=Pull%20Request&field2=subprojects&compare2=93&value2=Tpetra&field3=testsfailed&compare3=43&value3=0&filtercombine=and)

**Comment by bartlettroscoe** [comment]
FYI: Between these random failures, the build errors in #10842 (which have been fixed the last two days), and other errors like listed in #10858, no PRs that include changes to Tpetra have merged in a month since the merge commit 6c842cb13d9:

```
* 6c842cb13d9 "Merge Pull Request #10706 from brian-kelley/Trilinos/SpeedupAdditiveSchwarz"
| Author: trilinos-autotester <trilinos@sandia.gov>
| Date:   Thu Jul 14 17:45:09 2022 -0600 (4 weeks ago)
| 
|  100.0% packages/tpetra/core/src/
```

**Comment by bartlettroscoe** [comment]
CC: @trilinos/tpetra, @csiefer2 

As per the process documented in [Addressing Test Failures](https://github.com/trilinos/Trilinos/wiki/Addressing-test-failures), with the merge of PR #10871 disabling these tests (step **B.3**), I have added the `Disabled Tests` label (step **B.5**) and I am leaving this issue open (step **B.6**) to allow Tpetra developers to reproduce (step **B.7**) and fix these failures (step **B.8**).
